tests/pmap_test.py

```diff
@@ -280,6 +280,106 @@ def testShardedDeviceTuple(self):
     w = jit(lambda x: list(x)[0])(y)
     self.assertAllClose(w, x, check_dtypes=False)
 
+  @jtu.skip_on_devices("cpu", "gpu")
+  def testCollectivePermute(self):
+    device_count = xla_bridge.device_count()
+    if device_count != 2:
+      raise SkipTest("skipping because device_count != 2")
+
+    f = lambda x: lax.ppermute(x, perm=[(0, 1), (1, 0)], axis_name='i')
+    f = pmap(f, 'i')
+
+    x = np.arange(4 * device_count).reshape((device_count, 4))
+    ans = f(x)
+    expected = x[::-1]
+    self.assertAllClose(ans, expected, check_dtypes=False)
+
+  @jtu.skip_on_devices("cpu", "gpu")
+  def testRule30(self):
+    device_count = xla_bridge.device_count()
+    if device_count != 2:
+      raise SkipTest("skipping because device_count != 2")
+
+    def update_board(board):
+      # rule 30: https://en.wikipedia.org/wiki/Rule_30
```

Comment:
Move this comment to the top of the test and perhaps add a comment describing the point of the test (to test collective permute)

Reply:
Done.